src/dashboard/src/components/helpers.py

```diff
@@ -329,3 +330,42 @@ def stream_file_from_storage_service(url, error_message='Remote URL returned {}'
             'message': error_message.format(stream.status_code)
         }
         return json_response(response, status_code=400)
+
+
+def completed_units_efficient(unit_type='transfer', include_failed=True):
+    """Returns a list of unit UUIDs corresponding to the non-hidden completed
+    units. Uses a single database request. If ``include_failed`` is ``True``,
+    failed units will be included in the returned list. Note that the criteria
+    used here for determining that a unit is completed (and failed) are
+    intended to coincide exactly with the criteria used by
+    api/views.py::get_unit_status
+    """
+    table_name = 'Transfers'
+    pk_name = 'transferUUID'
+    if unit_type != 'transfer':
+        table_name = 'SIPs'
+        pk_name = 'sipUUID'
+    q = ('SELECT u.{pk_name}, j.jobType, j.microserviceGroup'
+         ' FROM {table_name} as u'
+         ' INNER JOIN Jobs as j'
+         ' ON j.SIPUUID = u.{pk_name}'
+         ' WHERE u.hidden = 0'
+         ' ORDER BY u.{pk_name}, j.createdTime DESC, j.createdTimeDec DESC;'.format(
+             table_name=table_name, pk_name=pk_name))
```

Comment:
In terms of style guides, we have this: https://wiki.archivematica.org/Contribute_code#Code_Style_Guide_For_Archivematica